### PR TITLE
LibraryPanels: Disconnect before connect during dashboard save

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR makes sure that only the current added libarary panels are connected to the dashboard.

Scenario: A user removes a library panel from a dashboard and saves the dashboard. We want to make sure that the removed libarary panel is disconnected. This PR takes care of this. This approach means that the last saved dashboard will win if there are several users trying to update the same dashboard. If we need a more complex solution let me know.

**Which issue(s) this PR fixes**:
Relates #29610

**Special notes for your reviewer**:
I also renamed `AvatarUrl` to `AvatarURL`.

### CODE REVIEW

**Review by marefr — COMMENTED** [review]
Nit. Would suggest keeping the rename in a separate PR to ease review

I was looking at mobile and seemed like a lot of changes. But I might be wrong so please disregard in that case

**Comment by hugohaggmark** [comment]
> Nit. Would suggest keeping the rename in a separate PR to ease review

Fixed!

**Review by kaydelaney — APPROVED** [review]

File: pkg/services/librarypanels/database.go [hunk 1]
```diff
@@ -115,6 +115,10 @@ func (lps *LibraryPanelService) connectDashboard(c *models.ReqContext, uid strin
 // connectLibraryPanelsForDashboard adds connections for all Library Panels in a Dashboard.
 func (lps *LibraryPanelService) connectLibraryPanelsForDashboard(c *models.ReqContext, uids []string, dashboardID int64) error {
 	err := lps.SQLStore.WithTransactionalDbSession(context.Background(), func(session *sqlstore.DBSession) error {
```
Comment (marefr) [inline pkg/services/librarypanels/database.go:117]: Something for the future to consider. Using `context.Background()` is only suggested when you don't have any other context. In this case you could use the request context `c.Context.Req.Context()` I think.

**Review by marefr — APPROVED** [review]
LGTM

File: pkg/services/librarypanels/database.go (hunk 1 above)
Comment (hugohaggmark) [reply pkg/services/librarypanels/database.go:117]: Good catch! I'll look into it and change.

**Review by torkelo — APPROVED** [review]
Looks good :)

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**